File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.py
class Solution(object):
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        n = len(nums)
    
        nums_set = set(nums)

        x = []
        
        for i in range(1, n+1):
            if(i not in nums):
                x.append(i)
                
        return x

File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.java
import java.util.*;

class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        
        int n = nums.length;
        
        List<Integer> x =new ArrayList<Integer>();
        
        Set<Integer> numSet = new HashSet<>();
        for (int num : nums) {
            numSet.add(num);
        }
        
        for(int i = 1; i <= n; i++){
            if(!numSet.contains(i)){
                x.add(i);
            }
        }
        return x;
    }
}